great_expectations/zep/type_lookup.py

```diff
@@ -22,24 +47,43 @@ class TypeLookup(
 
     def __init__(
         self,
-        __dict: Optional[Mapping[Hashable, Hashable]] = None,
+        __dict: Optional[Mapping[ValidTypes, ValidTypes]] = None,
         **kwargs: Hashable,
     ):
         __dict = __dict or {}
+        self._original_keys: Set[ValidTypes] = set()
```

Comment:
I think we should get rid of this notion of `_original_keys`. We end up maintaining it so it always matches `__dict`. I don't know if anyone cares if they originally added the `str` -> `type` mapping or if they added the `type` -> `str` mapping for a particular `str`, `type` pair.  I think it will be more useful to get the `str` keys or the `type` keys rather than the original keys.

If we are going to maintain 2 data structures, having a `__dict` with these hybrid data types and having `_original_keys` seems complicated. I believe it would be more understandable to have 2 dicts in this case: `str` -> `type` and a `type` -> `str`.

In terms of iterating twice. There just isn't too many of these that I think it will matter. That is it's unclear to me if we call iterating through everything is slower than the hash lookup we do in `original_items`.

Reply:
I may not be following everything here.

We don't allow overwrite of existing keys with `__setitem__`
The reason I added `_original_keys` is to prevent doing an overwrite of existing keys during the transaction commit/`.update()`.

I agree we could achieve the same thing by maintaining a mapping of just the `str`'s or just the `types` but that seems more complicated to maintain. Wouldn't that require extra conditionals as part of `__setitem__` or some slightly complicated dictionary comprehension to build it on the fly?

Reply:
Removed `original_keys` stuff.